### PR TITLE
Update neighbours

### PR DESCRIPTION
**Pull request by corradio**
PR required to be able to merge the split of zones.

The `ZONE_NEIGHBOURS` object now defines the graph over which zones can be flowtraced.
It therefore only includes zone pairs that:
- have an *exchange* parser (not just any parser!)
- do not have subzones (as parents are excluded from the flowtracing)

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
This might be slightly off topic but is the parent zones (zones with subkeys) used in the backend?
I haven't really seen a place where they are used in the frontend so instead of testing if they have subzone keys it might be simpler to just remove the parent zones?

If that is not possible can we at least clean them up by removing the capacities and/or parsers?

File: electricitymap/contrib/config/__init__.py [hunk 1]
```diff
@@ -173,6 +44,27 @@ def _can_zones_be_neighbours(
         for sub_zone_id in zone_config["subZoneNames"]:
             ZONE_PARENT[sub_zone_id] = zone_id
 
+# This object represents the edges of the flow-tracing graph
+ZONE_NEIGHBOURS: Dict[ZoneKey, List[ZoneKey]] = {}
```
Comment (pierresegonne) [inline electricitymap/contrib/config/__init__.py:48]: Can you replicate the structure I laid out with a generation function? 

This allows to unit test it and document more easily what the expected zone_neighbours object we get depending on the provided config file.

**Comment by corradio** [comment]
> This might be slightly off topic but is the parent zones (zones with subkeys) used in the backend? I haven't really seen a place where they are used in the frontend so instead of testing if they have subzone keys it might be simpler to just remove the parent zones?

Yes it is used in the backend in order to be able to make country aggregates. I also foresee it to be used by the frontend when we give the ability to toggle between country and regional view, but the team team has more knowledge than me :)

> If that is not possible can we at least clean them up by removing the capacities and/or parsers?

We have seen cases where we want to collect both parent and subzone data for experimentation. For example, we might want to have collected SE subzone data alongside SE for a while, before making the final switch.
For capacities that's another story - I'll let the team comment here (or consider cleaning things up in another PR).

File: electricitymap/contrib/config/__init__.py (hunk 1 above)
Comment (corradio) [reply electricitymap/contrib/config/__init__.py]: Done. I did not unit test it though.

Comment (corradio) [reply electricitymap/contrib/config/__init__.py]: Done

**Review by pierresegonne — COMMENTED** [review]
Could you add a simple unit test for the function? I can take over if you don't have tiem

**Comment by corradio** [comment]
> Could you add a simple unit test for the function? I can take over if you don't have tiem

would be terrific if you could add a few tests!

**Comment by pierresegonne** [comment]
> would be terrific if you could add a few tests!

Will do

**Comment by pierresegonne** [comment]
Should be tested now :)

**Review by pierresegonne — APPROVED** [review]
Ready to go!¬

**Pull request merged**